cmd/gokr-packer/packer.go

```diff
@@ -298,6 +294,93 @@ func findEnvFiles() (map[string]string, error) {
 	return contents, nil
 }
 
+func addToFileInfo(parent *fileInfo, path string) (error, time.Time) {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, time.Time{}
+		}
+		return err, time.Time{}
+	}
+
+	var latestTime time.Time
+	for _, entry := range entries {
+		// get existing file info
+		var fi *fileInfo
+		for _, ent := range parent.dirents {
+			if ent.filename == entry.Name() {
+				fi = ent
+				break
+			}
+		}
+
+		info, err := entry.Info()
+		if err != nil {
+			return err, time.Time{}
+		}
+
+		if latestTime.Before(info.ModTime()) {
+			latestTime = info.ModTime()
+		}
+
+		// or create if not exist
+		if fi == nil {
+			fi = &fileInfo{
+				filename: entry.Name(),
+				mode:     info.Mode(),
+			}
+			parent.dirents = append(parent.dirents, fi)
+		} else {
+			// file overwrite is not supported -> return error
+			if !entry.IsDir() || fi.fromHost != "" || fi.fromLiteral != "" {
+				return fmt.Errorf("file already exists in filesystem: %s", filepath.Join(path, entry.Name())), time.Time{}
+			}
+		}
+
+		// add content
+		if entry.IsDir() {
+			err, modTime := addToFileInfo(fi, filepath.Join(path, entry.Name()))
+			if err != nil {
+				return err, time.Time{}
+			}
+			if latestTime.Before(modTime) {
+				latestTime = modTime
+			}
+		} else {
+			fi.fromHost = filepath.Join(path, entry.Name())
+		}
+	}
+
+	return nil, latestTime
+}
+
+func findExtraFiles() (map[string]*fileInfo, error) {
+	buildPackages := buildPackagesFromFlags()
+	extraFiles := make(map[string]*fileInfo, len(buildPackages))
+	for pkg := range buildPackages {
+		path := filepath.Join("extrafiles", pkg)
```

Comment:
This code path is missing a warning message when the extrafiles directory does not match any packages. See line 279 for example.

Reply:
Yeah I already thought about this but it seems not so easy as for the other entries.
For the other options it is possible to find all existing config files and then compare them to the list of packages.

Because the extra files also supports sub directories it would require a more complex check.
The only option I would see currently is to build a string list of all path to all entries in the extrafiles directory and check if all of them starts with known package.

I am not sure if this is worth the effort. What do you think?

Reply:
That’s fair. We can figure this out later.